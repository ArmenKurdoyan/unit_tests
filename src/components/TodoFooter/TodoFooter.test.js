import { render, screen, fireEvent } from "@testing-library/react";
import TodoFooter from "./TodoFooter";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

const MockTodoFooter = ({ numberOfIncompleteTasks }) => {
  return (
    <BrowserRouter>
      <TodoFooter numberOfIncompleteTasks={numberOfIncompleteTasks} />
    </BrowserRouter>
  );
};

describe("Todo footer", () => {
  it('should render "task" when the number of incomplete tasks is one', async () => {
    render(<MockTodoFooter numberOfIncompleteTasks={1} />);
    const paragraphElement = screen.getByText(/1 task left/i);
    expect(paragraphElement).toBeInTheDocument();
  });

  it('should render "task" when the number of incomplete tasks is one', async () => {
    render(<MockTodoFooter numberOfIncompleteTasks={4} />);
    const paragraphElement = screen.getByText(/4 tasks left/i);
    expect(paragraphElement).toBeInTheDocument();
  });

  it('Should check snapshot test with 4 tasks', () => {
    const tree = renderer.create(<MockTodoFooter numberOfIncompleteTasks={4} />).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('Should check snapshot test with 0 tasks', () => {
    const tree = renderer.create(<MockTodoFooter numberOfIncompleteTasks={0} />).toJSON();

    expect(tree).toMatchSnapshot();
  })
});

// it('', async () => {
//   render(
//     <MockTodoFooter
//       numberOfIncompleteTasks={1}
//     />
//   )
//   const paragraphElement = screen.getByTestId("para");
//   expect(paragraphElement).toHaveTextContent("1 task left")
// });

// it('', async () => {
//   render(
//     <MockTodoFooter
//       numberOfIncompleteTasks={1}
//     />
//   )
//   const paragraphElement = screen.getByTestId("para");
//   expect(paragraphElement).toHaveTextContent("1 task left")
// });

// it('', async () => {
//   render(
//     <MockTodoFooter
//       numberOfIncompleteTasks={1}
//     />
//   )
//   const paragraphElement = screen.getByTestId("para");
//   expect(paragraphElement.textContent).toBe("1 task left")
// });
