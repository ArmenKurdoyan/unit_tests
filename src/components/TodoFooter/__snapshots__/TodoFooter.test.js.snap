// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo footer Should check snapshot test with 0 tasks 1`] = `
<div
  className="todo-footer"
>
  <p
    data-testid="para"
  >
    0
     
    tasks
     left
  </p>
  <a
    href="/followers"
    onClick={[Function]}
  >
    Followers
  </a>
</div>
`;

exports[`Todo footer Should check snapshot test with 4 tasks 1`] = `
<div
  className="todo-footer"
>
  <p
    data-testid="para"
  >
    4
     
    tasks
     left
  </p>
  <a
    href="/followers"
    onClick={[Function]}
  >
    Followers
  </a>
</div>
`;
